src/Castle.Windsor/MicroKernel/Arguments.cs

```diff
@@ -110,13 +92,17 @@ bool IDictionary.IsReadOnly
 			get { return arguments.IsReadOnly; }
 		}
 
+		public static Arguments Empty { get; } = new Arguments();
+
 		public void Add(object key, object value)
 		{
+			EnsureReadOnly();
```

Comment:
I think `EnsureReadOnly` is the opposite of what this method does which is why I suggested `EnsureWritable`.